### ISSUE TITLE
ESLint check passes even though problems are detected

### ISSUE THREAD

**Issue by DNin01** [ScratchAddons/ScratchAddons#7972]
Follow-up to #7134

@Hans5958

### Problem

[This](https://github.com/ScratchAddons/ScratchAddons/actions/runs/12102487462/job/33743623852) should have failed because problems were detected, but it was reported as successful anyway.

### Subject

https://github.com/ScratchAddons/ScratchAddons/blob/731445442b9c8dcc5b1174fb2604691120c8129a/.github/workflows/eslint.yml#L29-L33

### Solutions

Try:
- Removing `continue-on-error: true`
- Adding the command line flag `--max-warnings=0`

### Subtasks

- [x] [Address the reported problems](https://github.com/ScratchAddons/ScratchAddons/issues/7973) first so that contributors don't start getting errors they can't fix on up-to-date branches.

**Comment by Hans5958** [comment]
It has been that way since the creation of the workflow by Apple. I would guess that it is because she wanted to just check and not errors out.

**Comment by WorldLanguages** [comment]
Yes, it is a good idea to have it fail on error, but we first need to fix the existing problems.

**Comment by DNin01** [comment]
[The last run of ESLint](https://github.com/ScratchAddons/ScratchAddons/actions/runs/18137195844/job/51618895094) was silent and successful while the other runs logged errors, so there must not be any errors or warnings on the `master` branch.
